src/System.Net.Http/tests/FunctionalTests/IdnaProtocolTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Net.Test.Common;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public abstract class IdnaProtocolTests : HttpClientTestBase
```

Comment:
As abstract, this won't test WinHttpHandler or CurlHandler; is that intentional?

Reply:
There's a derived handler at the bottom of the file for WinHttpHandler/CurlHandler.  As Priya pointed out, this should probably be called HttpClientHandler_IdnaProtocolTests; its currently called DefaultHandler_IdnaProtocolTests, but I'll fix.

Also, as Priya pointed out above, this pattern is inconsistent with current practice; we currently have all the derived ManagedHandler_*Tests classes in ManagedHandlerTest.cs, and the base classes are not abstract.  I don't love this pattern.  I'd like to see us do something like some of the sockets tests do, where the pattern is an abstract base and derived classes for each variation.  This makes it easier to customize behavior per-variation, which I suspect we will want more of in the future, beyond just UseManagedHandler = true.

That said, I'm happy to revert to the existing pattern for now if that's what folks think is best.